operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/DefaultEventHandler.java

```diff
@@ -108,13 +109,28 @@ public void handleEvent(Event event) {
     try {
       lock.lock();
       log.debug("Received event: {}", event);
-      eventBuffer.addEvent(event);
-      executeBufferedEvents(event.getRelatedCustomResourceUid());
+
+      if (event.getRelatedCustomResourceUid() != null) {
+        eventBuffer.addEvent(event.getRelatedCustomResourceUid(), event);
+        executeBufferedEvents(event.getRelatedCustomResourceUid());
+      } else if (event.getCustomResourcesSelector() != null) {
+        for (String uid :
+            eventSourceManager.getLatestResourceUids(event.getCustomResourcesSelector())) {
+          eventBuffer.addEvent(uid, event);
+          executeBufferedEvents(uid);
+        }
+      }
     } finally {
       lock.unlock();
     }
   }
 
+  private void executeBufferedEvents(List<String> customResourceUids) {
```

Comment:
Both versions of the method should be merged into a single one, imo.

Reply:
this is a leftover, I'll remove it

Reply:
done